swift/ql/lib/codeql/swift/elements/KeyPathComponent.qll

```diff
@@ -1,4 +1,38 @@
-// generated by codegen/codegen.py, remove this comment if you wish to edit this file
 private import codeql.swift.generated.KeyPathComponent
 
-class KeyPathComponent extends Generated::KeyPathComponent { }
+class KeyPathComponent extends Generated::KeyPathComponent {
+  /**
+   * Property access like `.bar` in `\Foo.bar`.
+   */
+  predicate is_property() { getKind() = 3 }
```

Comment:
We use camelCase for predicates in QL (see https://github.com/github/codeql/blob/main/docs/ql-style-guide.md#naming).

Reply:
Sorry, I took

> could we add `is_property : predicate`, `is_subscript : predicate`, etc.

too literally. Done!